PIL/Image.py

```diff
@@ -586,14 +586,17 @@ def _dump(self, file=None, format=None):
     def __eq__(self, other):
         if self.__class__.__name__ != other.__class__.__name__:
             return False
-        a = (self.mode == other.mode)
-        b = (self.size == other.size)
-        c = (self.getpalette() == other.getpalette())
-        d = (self.info == other.info)
-        e = (self.category == other.category)
-        f = (self.readonly == other.readonly)
-        g = (self.tobytes() == other.tobytes())
-        return a and b and c and d and e and f and g
+
+        properties = ['mode', 'size', 'info', 'category', 'readonly']
+
+        props_eq = all(self.__getattribute__(p) == other.__getattribute__(p)
+                       for p in properties)
+        if not props_eq:
+            return False
```

Comment:
The build "failure" is from coverage, [because this `return False` line isn't hit](https://coveralls.io/builds/6660498/source?filename=PIL%2FImage.py#L595) by any tests.

Not a huge thing, but would be good to make sure this is really working for all Python versions. Please could you add a unit test that should return false here when comparing two images? 

Somewhere in here: https://github.com/python-pillow/Pillow/blob/master/Tests/test_image.py